examples_tests/51.RadixSort/main.cpp

```diff
@@ -0,0 +1,269 @@
+
+	SBufferRange<IGPUBuffer> scratch_gpu_range = { 0 };
+	scratch_gpu_range.size = in_gpu_range.size;
+	scratch_gpu_range.buffer = driver->createDeviceLocalGPUBufferOnDedMem(in_gpu_range.size);
+
+	SBufferRange<IGPUBuffer> histogram_gpu_range = { 0 };
+	histogram_gpu_range.size = radix_sort_dispatch_info.histogram_count * sizeof(uint32_t);
+	histogram_gpu_range.buffer = driver->createDeviceLocalGPUBufferOnDedMem(histogram_gpu_range.size);
+
+	for (uint32_t pass = 0; pass < RadixSortClass::PASS_COUNT; ++pass)
+	{
+		RadixSortClass::updateParameters(&radix_sort_push_constants, pass);
+
+		(pass == 0u)
+			? RadixSortClass::updateDescriptorSet(ds_histogram, { (pass % 2) ? scratch_gpu_range : in_gpu_range, histogram_gpu_range }, driver)
+			: RadixSortClass::updateDescriptorSet(ds_histogram, { (pass % 2) ? scratch_gpu_range : in_gpu_range }, driver);
+
+		driver->bindComputePipeline(histogram_pipeline);
+		driver->bindDescriptorSets(video::EPBP_COMPUTE, histogram_pipeline->getLayout(), 0u, 1u, &ds_histogram, nullptr);
+		RadixSortClass::dispatchHelper(histogram_pipeline->getLayout(), radix_sort_push_constants, radix_sort_dispatch_info, driver);
+
+		ExclusiveSumScan(driver, histogram_gpu_range.buffer, radix_sort_dispatch_info.histogram_count, ds_upsweep, upsweep_pipeline,
+			ds_downsweep, downsweep_pipeline);
+
+		(pass == 0u)
+			? RadixSortClass::updateDescriptorSet(ds_scatter, { (pass % 2) ? scratch_gpu_range : in_gpu_range, (pass % 2) ? in_gpu_range : scratch_gpu_range, histogram_gpu_range }, driver)
+			: RadixSortClass::updateDescriptorSet(ds_scatter, { (pass % 2) ? scratch_gpu_range : in_gpu_range, (pass % 2) ? in_gpu_range : scratch_gpu_range }, driver);
+
+		driver->bindComputePipeline(scatter_pipeline);
+		driver->bindDescriptorSets(video::EPBP_COMPUTE, scatter_pipeline->getLayout(), 0u, 1u, &ds_scatter, nullptr);
+		RadixSortClass::dispatchHelper(scatter_pipeline->getLayout(), radix_sort_push_constants, radix_sort_dispatch_info, driver);
+	}
+}
+
+int main()
+{
+	nbl::SIrrlichtCreationParameters params;
+	params.Bits = 24;
+	params.ZBufferBits = 24;
+	params.DriverType = video::EDT_OPENGL;
+	params.WindowSize = dimension2d<uint32_t>(512, 512);
+	params.Fullscreen = false;
+	params.Vsync = true;
+	params.Doublebuffer = true;
+	params.Stencilbuffer = false;
+	params.StreamingDownloadBufferSize = 0x10000000u; // 256MB download required
+	auto device = createDeviceEx(params);
+
+	if (!device)
+		return 1;
+
+	IVideoDriver* driver = device->getVideoDriver();
+
+	io::IFileSystem* filesystem = device->getFileSystem();
+	asset::IAssetManager* am = device->getAssetManager();
+
+	unsigned int seed = 128;
+
+	// Create (an almost) 256MB input buffer
+	const size_t in_count = (1 << 25) - 23;
+	const size_t in_size = in_count * sizeof(SortElement);
+
+	std::cout << "Input element count: " << in_count << std::endl;
+
+	SortElement* in = new SortElement[in_count];
+	srand(seed++);
+	for (size_t i = 0u; i < in_count; ++i)
+	{
+		in[i].key = rand();
```

Comment:
`srand` and `rand` are passe

Since C++11 mersenne twister from the standard library should be the default PRNG that you should use.